Program.cs

```diff
@@ -1,7 +1,25 @@
 ﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+   internal class Program {
+      static void Main (string[] args) {
+         Console.Write ("Enter a positive number: ");
+         if (int.TryParse (Console.ReadLine (), out int n) && n >= 0)
+            Console.Write ($"Factorial of {n} is {Factorial (n)}.");
+         else Console.WriteLine ("Invalid character. Enter a positive integer.");
+      }
+      static int Factorial (int n) {
+         //works upto 12!
+         int fact = 1;
+         for (int i = 1; i <= n; i++)
+            fact *= i;

```

Comment:
You could avoid this computation for n (0, 1 and 2) by defining certain conditions before the loop itself. 
That avoids, unnecessary loop initialization, increments and computation for these inputs.

Reply:
Used condition for n(0,1 and 2).